### PR TITLE
Feat: cloud deployment + multiple service centers

### PR DESCRIPTION
**Pull request by thecampagnards**

### CODE REVIEW

File: client/src/components/projects/project/project.page.jsx [hunk 1]
```diff
@@ -123,7 +123,7 @@ export class ProjectComponent extends React.Component {
     sourceCode: Joi.boolean().label('Source Code'),
     specifications: Joi.boolean().label('Specifications'),
     projectManager: Joi.string().trim().alphanum().empty('').label('Project Manager'),
-    serviceCenter: Joi.string().trim().alphanum().empty('').label('Service Center'),
+    serviceCenter: Joi.array().empty('').label('Service Center'),
```
Comment (Ezian) [inline client/src/components/projects/project/project.page.jsx]: ‼️ I don't think it's the proper way to validate an array of string:

~~~js
serviceCenter: Joi.array().items(Joi.string().trim().alphanum()).label('Service Center'),
~~~

❓ I don't know what is the expected behavior of the `empty` part. Could you explain it to me ?

File: server/controllers/projects.go [hunk 2]
```diff
@@ -238,11 +242,17 @@ func validateEntities(entityRepo types.EntityRepo, projectToSave, projectFromDB
 	}
 
 	// If a service center is provided, check it exists in the entity collection
-	if statusCode, errMessage := validateEntity(entityRepo, projectToSave.ServiceCenter, projectFromDB.ServiceCenter, types.ServiceCenterType, authUser); errMessage != "" {
-		return statusCode, errMessage
+	var statusCode = http.StatusOK
+	var errMessage = ""
+	for _, sCSave := range projectToSave.ServiceCenter {
+		for _, sCDB := range projectFromDB.ServiceCenter {
+			if statusCode, errMessage = validateEntity(entityRepo, sCSave, sCDB, types.ServiceCenterType, authUser); errMessage == "" {
```
Comment (Ezian) [inline server/controllers/projects.go]: ⁉️  Strange code. If it work, you return. If it fail, you retry. But you return only the last error message.

What is expected ?

File: server/controllers/projects.go [hunk 3]
```diff
@@ -354,7 +364,7 @@ func (p *Projects) createProjectToSave(database *mongo.DadMongo, id string, proj
 	modifiedDetails := projectToSave.Name != existingProject.Name ||
 		strings.Join(projectToSave.Domain, ";") != strings.Join(existingProject.Domain, ";") ||
 		projectToSave.ProjectManager != existingProject.ProjectManager ||
-		projectToSave.ServiceCenter != existingProject.ServiceCenter ||
+		!reflect.DeepEqual(projectToSave.ServiceCenter, existingProject.ServiceCenter) ||
```
Comment (Ezian) [inline server/controllers/projects.go]: ⁉️ I am perplexe. Equality comparator is enough for array. Why using reflect ?

File: client/src/components/projects/project/matrix/matrix.component.jsx [hunk 4]
```diff
@@ -93,7 +93,7 @@ class Matrix extends React.Component {
         serviceNameCell,
         (<Table.Cell key='deployed'>
           <Form>
-            {(service.declarativeDeployement || isIsolatedNetwork) && !readOnly
+            {(service.declarativeDeployement || isIsolatedNetwork || isCloud) && !readOnly
```
Comment (Ezian) [inline client/src/components/projects/project/matrix/matrix.component.jsx]: ❓ Cloud is considered as Isolated network ? People declare their services ?

File: server/controllers/projects.go (hunk 3 above)
Comment (thecampagnards) [reply server/controllers/projects.go]: Yes the == doesn't work on that, that why I used reflect for the equality of two arrays.

File: client/src/components/projects/project/project.page.jsx (hunk 1 above)
Comment (thecampagnards) [reply client/src/components/projects/project/project.page.jsx]: mb

File: server/controllers/projects.go (hunk 2 above)
Comment (thecampagnards) [reply server/controllers/projects.go]: I clean this part on the last commit

**Pull request merged**